services/minespace-web/src/components/dashboard/mine/MineDashboard.js

```diff
@@ -59,17 +67,25 @@ export class MineDashboard extends Component {
   componentWillReceiveProps(nextProps) {
     const { activeTab, id } = nextProps.match.params;
     if (activeTab !== this.state.activeTab) {
-      this.setState({ activeTab });
+      this.setState({ activeTab: this.extractActiveTab(activeTab) });
     }
     if (!nextProps.staticContentLoadingIsComplete) {
       this.loadStaticContent();
     }
 
-    if (nextProps.match.params.id !== this.props.match.params.id) {
-      this.loadMine(id, activeTab);
+    if (
```

Comment:
I find this condition a little difficult to read- if we were to translate this condition into English, what does it _mean_? Since we're referencing `this.extractActiveTab(activeTab)` in both the condition and the code block body, can we use the state variable or call the function once and store it in a variable?